diff --git a/src/utility/amx.cpp b/src/utility/amx.cpp
--- a/src/utility/amx.cpp
+++ b/src/utility/amx.cpp
@@ -29,58 +29,7 @@ cell AMX_NATIVE_CALL Utility::hookedNative(AMX *amx, cell *params)
 
 int Utility::checkInterfaceAndRegisterNatives(AMX *amx, AMX_NATIVE_INFO *amxNativeList)
 {
-	AMX_HEADER *amxHeader = reinterpret_cast<AMX_HEADER*>(amx->base);
-	AMX_FUNCSTUBNT *amxNativeTable = reinterpret_cast<AMX_FUNCSTUBNT*>(amx->base + amxHeader->natives);
-	int amxRegisterResult = amx_Register(amx, amxNativeList, -1);
-	bool foundNatives = false;
-	bool hookedNatives = false;
-	int numberOfNatives = 0;
-	amx_NumNatives(amx, &numberOfNatives);
-	for (int i = 0; i < numberOfNatives; ++i)
-	{
-		char *name = reinterpret_cast<char*>(amx->base + amxNativeTable[i].nameofs);
-		if (std::string(name).find("Streamer_") != std::string::npos)
-		{
-			foundNatives = true;
-			if (!amxNativeTable[i].address)
-			{
-				Utility::logError("Obsolete or invalid native \"%s\" found (script might need to be recompiled with the correct include file).", name);
-				amxNativeTable[i].address = reinterpret_cast<cell>(hookedNative);
-				hookedNatives = true;
-			}
-		}
-	}
-	if (foundNatives)
-	{
-		cell amxAddr = 0;
-		int includeFileValue = 0;
-		if (!amx_FindPubVar(amx, "Streamer_IncludeFileVersion", &amxAddr))
-		{
-			cell *amxPhysAddr = NULL;
-			if (!amx_GetAddr(amx, amxAddr, &amxPhysAddr))
-			{
-				includeFileValue = static_cast<int>(*amxPhysAddr);
-			}
-		}
-		if (includeFileValue != INCLUDE_FILE_VERSION)
-		{
-			std::ostringstream includeFileVersion;
-			if (includeFileValue <= 0)
-			{
-				includeFileVersion << "unknown version";
-			}
-			else
-			{
-				includeFileVersion << std::hex << std::showbase << includeFileValue;
-			}
-			Utility::logError("Include file version (%s) does not match plugin version (%#x) (script might need to be recompiled with the correct include file).", includeFileVersion.str().c_str(), INCLUDE_FILE_VERSION);
-		}
-	}
-	if (hookedNatives)
-	{
-		amxRegisterResult = amx_Register(amx, amxNativeList, -1);
-	}
-	return amxRegisterResult;
+	return amx_Register(amx, amxNativeList, -1);
 }
 
 void Utility::destroyAllItemsInInterface(AMX *amx)
